app/src/main/java/com/arctouch/giphy/presentation/main/MainActivity.kt

```diff
@@ -0,0 +1,32 @@
+package com.arctouch.giphy.presentation.main
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.arctouch.giphy.R
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var viewModel: MainViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        setupViewModel()
+        getDataButton.setOnClickListener {
+            viewModel.loadData()
+        }
+    }
+
+    private fun setupViewModel() {
+        viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)
```

Comment:
For now, this is okay, but we're going to remove it from activity and let Dagger handle this.